edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/device/DeviceMessageHandler.cs

```diff
@@ -71,10 +71,13 @@ public Task ProcessMethodResponseAsync(IMessage message)
             return Task.CompletedTask;
         }
 
-        public void BindDeviceProxy(IDeviceProxy deviceProxy)
+        public void BindDeviceProxy(IDeviceProxy deviceProxy, Option<Action> initWhenBound)
         {
             this.underlyingProxy = Preconditions.CheckNotNull(deviceProxy);
-            this.connectionManager.AddDeviceConnection(this.Identity, this);
+
+            var connectionCreation = this.connectionManager.AddDeviceConnection(this.Identity, this);
```

Comment:
Can we wrap all of this async logic inside an async void method?

Reply:
moved out